modules/core/src/main/java/org/apache/ignite/internal/properties/IgniteProductVersion.java

```diff
@@ -29,11 +30,9 @@
  */
 public class IgniteProductVersion implements Serializable {
     private static final Pattern VERSION_PATTERN =
-            Pattern.compile("(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<maintenance>\\d+)((?<snapshot>-SNAPSHOT)|-(?<alpha>alpha\\d+))?");
+            Pattern.compile("(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<maintenance>\\d+)(\\.(?<patch>\\d+))?(-(?<preRelease>[0-9A-Za-z]+))?");
```

Comment:
it's better to use non-capturing groups, e.g.: `(?:\\.(?<patch>\\d+))`

Reply:
Why is it better?

Reply:
It's a minor optimization to save a little bit of memory

Reply:
I tried to fix it.

Reply:
Fixed it, I think